homeassistant/components/home_connect/button.py

```diff
@@ -0,0 +1,160 @@
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .common import setup_home_connect_entry
+from .const import APPLIANCES_WITH_PROGRAMS, DOMAIN
+from .coordinator import (
+    HomeConnectApplianceData,
+    HomeConnectConfigEntry,
+    HomeConnectCoordinator,
+)
+from .entity import HomeConnectEntity
+from .utils import get_dict_from_home_connect_error
+
+
+class HomeConnectCommandButtonEntityDescription(ButtonEntityDescription):
+    """Describes Home Connect button entity."""
+
+    key: CommandKey
+
+
+COMMAND_BUTTONS = (
+    HomeConnectCommandButtonEntityDescription(
+        key=CommandKey.BSH_COMMON_OPEN_DOOR,
+        translation_key="open_door",
+    ),
+    HomeConnectCommandButtonEntityDescription(
+        key=CommandKey.BSH_COMMON_PARTLY_OPEN_DOOR,
+        translation_key="partly_open_door",
+    ),
+    HomeConnectCommandButtonEntityDescription(
+        key=CommandKey.BSH_COMMON_PAUSE_PROGRAM,
+        translation_key="pause_program",
+    ),
+    HomeConnectCommandButtonEntityDescription(
+        key=CommandKey.BSH_COMMON_RESUME_PROGRAM,
+        translation_key="resume_program",
+    ),
+)
+
+
+def _get_entities_for_appliance(
+    entry: HomeConnectConfigEntry,
+    appliance: HomeConnectApplianceData,
+) -> list[HomeConnectEntity]:
+    """Get a list of entities."""
+    entities: list[HomeConnectEntity] = []
+    entities.extend(
+        HomeConnectCommandButtonEntity(entry.runtime_data, appliance, description)
+        for description in COMMAND_BUTTONS
+        if description.key in appliance.commands
+    )
+    if appliance.info.type in APPLIANCES_WITH_PROGRAMS:
+        entities.append(
+            HomeConnectStopProgramButtonEntity(entry.runtime_data, appliance)
+        )
+
+    return entities
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: HomeConnectConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Home Connect select entities."""
+    setup_home_connect_entry(
+        entry,
+        _get_entities_for_appliance,
+        async_add_entities,
+    )
+
+
+class HomeConnectButtonEntity(HomeConnectEntity, ButtonEntity):
+    """Describes Home Connect button entity."""
+
+    entity_description: ButtonEntityDescription
+
+    def __init__(
+        self,
+        coordinator: HomeConnectCoordinator,
+        appliance: HomeConnectApplianceData,
+        desc: ButtonEntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(
+            coordinator,
+            appliance,
+            # The entity is subscribed to the appliance connected event,
+            # but it will receive also the disconnected event
+            ButtonEntityDescription(
```

Comment:
Side note: I think we should refactor how we build the coordinator context and instead have a specific parameter `context` for that in the `HomeConnectEntity`. It's confusing to pass an entity description and then not use that as the entity description.

Then we could also refactor this platform and have a single entity class and let the entity description describe how to subscribe for updates, ie what coordinator context to set, and how the press action should call the client.